.github/workflows/reusable-e2e.yml

```diff
@@ -0,0 +1,320 @@
+on:
+  workflow_call:
+    inputs:
+      commit:
+        # commit to run tests against
+        required: true
+        type: string
+      repository:
+        # repository to run tests against (possibly a fork of amplify-ui)
+        required: true
+        type: string
+    secrets:
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      DOMAIN:
+        required: true
+      PHONE_NUMBER:
+        required: true
+      USERNAME:
+        required: true
+      NEW_PASSWORD:
+        required: true
+      VALID_PASSWORD:
+        required: true
+
+jobs:
```

Comment:
Below is copy + paste of e2e test codes we already had. 

The only difference is that now we're using `inputs.commit` and `inputs.repository` that we defined above.